File: chat-sdk/src/main/java/ru/usedesk/chat_sdk/external/entity/UsedeskMessageButtons.java
package ru.usedesk.chat_sdk.external.entity;

import java.util.ArrayList;
import java.util.List;

public class UsedeskMessageButtons {
    private final String messageText;
    private final List<UsedeskMessageButton> messageButtons = new ArrayList<>();

    UsedeskMessageButtons(String messageText) {
        if (messageText != null) {
            String tempText = messageText;
            while (tempText.contains("{{button:") && tempText.contains("}}")) {
                int start = tempText.indexOf("{{button:");
                int end = tempText.indexOf("}}", start);

                //Выделим секцию кнопки
                String buttonText = tempText.substring(start, end + 2);

                UsedeskMessageButton messageButton = UsedeskMessageButton.create(buttonText);

                if (messageButton != null) {
                    String replaceBy;
                    if (messageButton.isShow()) {
                        replaceBy = messageButton.getText();
                    } else {
                        replaceBy = "";
                    }
                    //Удалим её из исходного сообщения
                    messageText = messageText.replace(buttonText, replaceBy);

                    messageButtons.add(messageButton);
                }

                tempText = tempText.substring(end);
            }
        }

        this.messageText = messageText;
    }

    public String getMessageText() {
        return messageText;
    }

    public List<UsedeskMessageButton> getMessageButtons() {
        return messageButtons;
    }

}

File: chat-sdk/src/main/java/ru/usedesk/chat_sdk/external/entity/old/UsedeskMessageButtons.kt
package ru.usedesk.chat_sdk.external.entity.old

class UsedeskMessageButtons internal constructor(
        val messageText: String
) {
    private val messageButtons: MutableList<UsedeskMessageButton> = mutableListOf()

    fun getMessageButtons(): List<UsedeskMessageButton> {
        return messageButtons
    }

    init {
        var text = messageText
        while (text.contains("{{button:") && text.contains("}}")) {
            val start = text.indexOf("{{button:")
            val end = text.indexOf("}}")

            //Выделим секцию кнопки
            val buttonText = text.substring(start, end + 2)

            //Удалим её из исходного сообщения
            text = text.replace(buttonText, "")
            val messageButton = UsedeskMessageButton(buttonText)
            if (!messageButton.isShow) {
                text = text.replace(messageButton.text, "")
            }
            messageButtons.add(messageButton)
        }
    }
}